### DISCUSSION TITLE
Using TypeVar with Annotated Depends

### DISCUSSION THREAD

**Discussion by 9en9i** [fastapi/fastapi#9450] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated, TypeVar

from fastapi import FastAPI, Depends

app = FastAPI()

T = TypeVar("T")

Inject = Annotated[T, Depends()]


class FirstService:
    pass


class SecondService:
    pass


@app.get("/first")
async def first_service(service: Inject[FirstService]):
    pass


@app.get("/second")
async def second_service(service: Inject[SecondService]):
    pass


if __name__ == "__main__":
    import uvicorn

    uvicorn.run("main:app", host="0.0.0.0", port=8000, reload=True)
```

### Description

When using TypeVar, dependencies are injected incorrectly.
To see this, just run my code in debug mode, put a breakpoint on line 27 (second_service endpoint), and call the /second endpoint in the swagger. Expected Result:
Service variable has type SecondService.
Instead we see type FirstService.

### Operating System

macOS

### FastAPI Version

0.95.1

### Python Version

Python 3.11.1

**Comment by nzig** [comment, marked as answer]
Interesting use-case.

This appears to be a bug. I've opened #9479 to fix it.

**Reply by tiangolo** [reply]
Thanks @nzig! :rocket: 

This fix is available in FastAPI 0.110.1 :tada:
